### DISCUSSION TITLE
Keep chat up / open a chat window

### DISCUSSION THREAD

**Discussion by katastic** [OpenRCT2/OpenRCT2#15675] [Feature requests]
It'd be nice to either keep the chat text open, or spawn a "chat window" so we don't have to keep hitting C to see the chats. Or at the very least, adjust the time before chats go away (so we can set it to something pretty high and fake it.)

ALSO. Long chats don't wrap around automatically which can be a problem for lower resolution users with long texts. 

I'd be willing to contribute to this if you want. But I don't know your codebase, so point me to the general area if you want.

Thanks for your time! 

Have a great day, and thanks for this amazing project. We've been having a blast all day.

**Comment by pizza2004** [comment]
I like the idea of being able to open a chat window by holding down on the multiplayer button.  This could also be used to allow the entire chat log for the time you've been connected to be stored and scrolled through (a function I'd desperately like, sometimes tabbing away to do other things and coming back to having missed an entire conversation) and allowing easier chatting as well as it were.  I would also possibly be willing to contribute to this, if someone just points out where we should start to make sure it's done properly.

**Comment by Krutonium** [comment]
I don't produce code for this project (directly, I don't know C), but perhaps start by looking at Chat.c?

**Comment by pizza2004** [comment]
Alright, I guess I'll start there.  I have a few ideas of what I might need to do from looking at the code already, but I've never made a new window before, and it seems just a tad daunting.  I guess perhaps the best thing to do is just jump right in and start changing things until I have something that works. :D
